nemo_curator/classifiers/aegis.py

```diff
@@ -0,0 +1,246 @@
+        raw_pred_column="_aegis_raw_pred",
+        keep_raw_pred=False,
+        max_chars=6000,
+        device_type="cuda",
+    ):
+        config = AegisConfig(peft_model_name_or_path=aegis_variant, token=token)
+
+        self.text_field = text_field
+        self.labels = AEGIS_LABELS
+        self.out_dim = len(self.labels)
+        self.raw_pred_column = raw_pred_column
+        self.keep_raw_pred = keep_raw_pred
+
+        model = AegisHFModel(config=config)
+
+        super().__init__(
+            model=model,
+            labels=self.labels,
+            filter_by=filter_by,
+            batch_size=batch_size,
+            out_dim=self.out_dim,
+            pred_column=pred_column,
+            max_chars=max_chars,
+            device_type=device_type,
+            autocast=False,
+        )
+
+    def _wrap_in_prompt(self, df):
+        documents = df[self.text_field].to_arrow().to_pylist()
+        prompts = [format_aegis(doc[: self.max_chars]) for doc in documents]
+        df["_hidden_text"] = cudf.Series(prompts)
+        return df
+
+    def _parse_response(self, raw_response: str) -> str:
+        lines = raw_response.split("\n")
+        if lines[0].strip() == "safe":
+            return "safe"
+        elif lines[0].strip() == "unsafe":
+            if len(lines) < 2:
+                return "unknown"
+            potential_label = lines[1].strip()
+            if potential_label not in AEGIS_LABELS[2:]:
+                return "unknown"
+
+            return potential_label
+        else:
+            return "unknown"
+
+    def _postprocess_responses(self, df):
+        tokenizer = self.model.load_tokenizer()
+        generated_tokens = df[self.raw_pred_column].to_arrow().to_pylist()
+        generated_tokens = tokenizer.batch_decode(
+            generated_tokens,
+            skip_special_tokens=True,
+        )
+        original_lengths = df["_hidden_text"].str.len().to_arrow().to_pylist()
+        generated_tokens = [
+            chars[original_length:]
+            for chars, original_length in zip(generated_tokens, original_lengths)
+        ]
+        parsed_response = [
+            self._parse_response(response) for response in generated_tokens
+        ]
+        if self.keep_raw_pred:
+            df[self.raw_pred_column] = cudf.Series(generated_tokens)
+        else:
+            df = df.drop(columns=[self.raw_pred_column])
+        df[self.pred_column] = cudf.Series(parsed_response)
+        return df
+
+    def _run_classifier(self, dataset: DocumentDataset):
+        print("Starting AEGIS classifier inference", flush=True)
+        ddf = dataset.df
+        hidden_meta = ddf._meta.copy()
+        hidden_meta["_hidden_text"] = "DUMMY_STRING"
+        ddf = ddf.map_partitions(self._wrap_in_prompt, meta=hidden_meta)
+        columns = ddf.columns.tolist()
+        pipe = op.Sequential(
+            op.Tokenizer(
+                self.model, cols=["_hidden_text"], tokenizer_type="sentencepiece"
```

Comment:
```suggestion
                self.model, cols=["_hidden_text"], tokenizer_type="default"
```

Reply:
I made the change, but what is this doing exactly?

Reply:
So there are mostly 3 flavors of tokenizers  in use today: 
- `subword` : Used by `bert` like models
-  `BPE` (Byte Pair Encoding) : Used  by GPT like models 
- `sentence-piece`: Another flavor of BPE, Used by models like `llama-3` etc. 

Today, only `subword` is GPU accelerated, in the future `BPE` and `SentencePiece` will be GPU accelerated but in future we shall accelerate all of them

https://github.com/rapidsai/crossfit/blob/97febffff4bbc32317afaae0aaa94b57d13645c6/crossfit/op/tokenize.py#L31-L34

What `default` does is (which is functionally same as `sentence-piece`) is use the provided tokenizer in the model directly.  See below: 

https://github.com/rapidsai/crossfit/blob/97febffff4bbc32317afaae0aaa94b57d13645c6/crossfit/op/tokenize.py#L59-L75

**Why it matters:** 
Even though there is no functionality difference b/w them today, it might be there  in the future. 
Having `detault` means that no unexpected changes happen here when we update `crossfit`.